0000-ledger-plugin-interface-2/0000-ledger-plugin-interface-2.md

```diff
@@ -0,0 +1,249 @@
+
+Create a new instance of the plugin. Each instance typically corresponds to a different ledger. However, some plugins MAY deviate from a strict one-to-one relationship and MAY internally act as a virtual connector to more than one counterparty.
+
+Throws `InvalidFieldsError` if the constructor is given incorrect arguments.
+
+###### Parameters
+| Name | Type | Description |
+|:--|:--|:--|
+| opts | <code>[PluginOptions](#class-pluginoptions)</code> | Object containing ledger-related settings. May contain plugin-specific fields. |
+
+###### Example
+```js
+const ledgerPlugin = new LedgerPlugin({
+
+  // auth parameters are defined by the plugin
+
+  _store: {
+    // persistence may be required for internal use by some ledger plugins
+    // (e.g. when the ledger has reduced memo capability and we can only put an ID in the memo)
+    // Store a value under a key
+    put: (key, value) => {
+      // Returns Promise.<null>
+    },
+    // Fetch a value by key
+    get: (key) => {
+      // Returns Promise.<Object>
+    },
+    // Delete a value by key
+    del: (key) => {
+      // Returns Promise.<null>
+    }
+  }
+})
+```
+
+For a detailed description of these properties, please see [`PluginOptions`](#class-pluginoptions).
+
+#### LedgerPlugin.version
+<code>LedgerPlugin.**version**:Number</code>
+
+Always `2` for this version of the Ledger Plugin Interface.
+
+### Connection Management
+
+#### LedgerPlugin#connect
+<code>ledgerPlugin.connect( options:[ConnectOptions](#class-connectoptions ) ⇒ Promise.&lt;null></code>
+
+`options` is optional.
+
+Initiate ledger event subscriptions. Once `connect` is called the ledger plugin MUST attempt to subscribe to and report ledger events. Once the connection is established, the ledger plugin should emit the [`connect`](#event-connect-) event. If the connection is lost, the ledger plugin SHOULD emit the [`disconnect`](#event-disconnect-) event.
+
+Rejects with `InvalidFieldsError` if credentials are missing, and `NotAcceptedError` if credentials are rejected.
+Rejects with `TypeError` if `options.timeout` is passed but is not a `Number`.
+
+#### LedgerPlugin#disconnect
+<code>ledgerPlugin.disconnect() ⇒ Promise.&lt;null></code>
+
+Unsubscribe from ledger events.
+
+#### LedgerPlugin#isConnected
+<code>ledgerPlugin.isConnected() ⇒ Boolean</code>
+
+Query whether the plugin is currently connected.
+
+#### Event: `connect`
+<code>ledgerPlugin.on('connect', () ⇒ )</code>
+
+Emitted whenever a connection is successfully established.
+
+#### Event: `disconnect`
+<code>ledgerPlugin.on('disconnect', () ⇒ )</code>
+
+Emitted when the connection has been terminated or lost.
+
+#### Event: `error`
+<code>ledgerPlugin.on('error', ( **err**:Error ) ⇒ )</code>
+
+General event for fatal exceptions. Emitted when the plugin experienced an unexpected unrecoverable condition. Once triggered, this instance of the plugin MUST NOT be used anymore.
+
+### Sending
+
+#### LedgerPlugin#sendData
```

Comment:
If you specifically wanted to implement a ledger plugin that did send ILP packets over conditional ledger transfers, would you just implement one that looks at the data being sent and handles it in a different way?